docs/src/content/docs/concepts/arrays.mdx

```diff
@@ -0,0 +1,228 @@
+4. Chains another `map` operation to call the `FetchAndSummarize` component for each document
+5. Calls `toArray()` to convert the `GsxArray<ArxivSummary>` into a `ArxivSummary[]`
+
+You can find the full code for this in the [Deep Research example](https://github.com/gensx-inc/gensx/tree/main/examples/).
+
+## Working with arrays
+
+Now that you've seen an example showing a lot of the functionality of `gsx.array`, let's explore each of these operations in more detail.
+
+### Creating an array
+
+To access the array operations, you first need to create a `GsxArray`. You can create an array from raw values:
+
+```tsx
+import { gsx } from "gensx";
+
+// Create from raw values
+const numbers = gsx.array<number>([1, 2, 3]);
+```
+
+You can also create arrays from components:
+
+```tsx
+const NumberWrapper = gsx.Component<{ n: number }, number>(
+  "NumberWrapper",
+  ({ n }) => n,
+);
+
+// Create from components
+const wrappedNumbers = gsx.array<NumberWrapper>([
+  <NumberWrapper n={1} />,
+  <NumberWrapper n={2} />,
+  <NumberWrapper n={3} />,
+]);
+```
+
+To convert a `GsxArray` back to a plain array, you simply call the `toArray()` method which produces an array of resolved component outputs.
+
+### Map and FlatMap
+
+`gsx.array` supports both `map` and `flatMap` operations. Both of these operations behave exactly as you'd expect if you're familiar with `array.map` and `array.flatMap` in JavaScript.
+
+#### Map
+
+Map transforms each element in an array using a component:
+
+```tsx
+const numbers = gsx.array([1, 2, 3]);
+
+const doubled = await numbers
+  .map<number>((n) => <NumberDoubler value={n} />)
+  .toArray();
+// Result: [2, 4, 6]
+```
+
+#### FlatMap
+
+Similarly, flatMap invokes a component for each element in the array and flattens the result.
+
+For example, the code below retrieves an array of search results for each query and flattens the results into a single array:
+
+```tsx
+const queries = gsx.array(["chain of thought", "reasoning models"]);
+
+const documents = await queries
+  .flatMap<SearchResult>((query) => (
+    <WebSearch query={query} /> // returns a list of documents
+  ))
+  .toArray();
+```
+
+### Filter
+
+Filters allow you to filter out elements that don't match a given condition. Inside the filter predicate, you can pass either:
+
+- A component that returns a boolean
+- A plain boolean expression
+
+#### Filtering with a component
+
+When you filter with a component, the component needs to result in a boolean value.
+
+```tsx
+// Define a component that returns a boolean
+const EvenNumberFilter = gsx.Component<{ value: number }, boolean>(
+  "EvenNumberFilter",
+  ({ value }) => {
+    return value % 2 === 0;
+  },
+);
+
+// Use a component that returns a boolean to filter
+const evenNumbers = await gsx
+  .array<number>([10, 11, 12, 13, 14])
+  .filter((n) => <EvenNumberFilter value={n} />)
+  .toArray();
```

Comment:
This example feels a little contrived. I don't think we would actually suggest that anyone create an `EvenNumberFilter` component. Probably this example would feel a little more natural if it was something along the lines of: 

```tsx
const numbers = [<DoubleNumber n={1}/>, <DoubleNumber n={2}/>, <DoubleNumber n={3}/>];

gsx.array(numbers).filter(n => n < 5);
```

Reply:
This particular example is demonstrating that a component that returns a boolean can be used in a filter expression so we'd need an example that fit's that criteria. We could do something more complex but it seems important that we show the component code inline to get the point across for this particular one.

Reply:
I'm fine leaving it as is, up to you. The example just struck me as a little odd, but we should probably just ship it and we can update it if we hear feedback from users that they have difficulty groking the concept.

Reply:
sounds good, will be easy to update in the future if needed.